### PR TITLE
bpo-42171: Add PEP573-related items to the limited API

### PR DESCRIPTION
**Pull request by encukou**
https://bugs.python.org/issue42171

Automerge-Triggered-By: GH:encukou

### CODE REVIEW

File: Include/methodobject.h [hunk 1]
```diff
@@ -89,11 +89,17 @@ PyAPI_FUNC(PyObject *) PyCMethod_New(PyMethodDef *, PyObject *,
...
+ *
+ * METH_METHOD_ARGS_KWD exports a specific combination of flags
+ * for the limited API.
  */
 
 #if !defined(Py_LIMITED_API) || Py_LIMITED_API+0 >= 0x03090000
 #define METH_METHOD 0x0200
 #endif
+#if !defined(Py_LIMITED_API) || Py_LIMITED_API+0 >= 0x03100000
+#define METH_METHOD_ARGS_KWD (METH_METHOD | 0x0080 | METH_KEYWORDS)
```
Comment (vstinner) [inline Include/methodobject.h]:
```suggestion
#define METH_METHOD_ARGS_KWD (METH_METHOD | METH_FASTCALL | METH_KEYWORDS)
```

**Review by vstinner — COMMENTED** [review]
Since PEP 590 has been approved, is there still a reason to exclude METH_FASTCALL from the limited C API? The limited C API is supposed to hide implementation details. Do we still consider that "fastcall" and "vectorcall" are implementation details?

Why not adding METH_FASTCALL to the limited C API, rather than adding a new METH_METHOD_ARGS_KWD macro?

File: PC/python3dll.c [hunk 2]
```diff
@@ -342,6 +342,7 @@ EXPORT_FUNC(PyModule_AddFunctions)
 EXPORT_FUNC(PyModule_AddIntConstant)
 EXPORT_FUNC(PyModule_AddObject)
 EXPORT_FUNC(PyModule_AddStringConstant)
+EXPORT_FUNC(PyModule_AddType)
```
Comment (vstinner) [inline PC/python3dll.c:345]: You may move these changes into a separated PR, since these changes are not controversial.

**Comment by encukou** [comment]
OK! If you're fine adding `METH_FASTCALL`, I have nothing against it :)

File: PC/python3dll.c (hunk 2 above)
Comment (encukou) [reply PC/python3dll.c:345]: They won't really help without the METH_METHOD calling convention, so I'll keep them here :)

**Review by vstinner — APPROVED** [review]
LGTM, thanks.

@markshannon wrote that "FASTCALL" is a bad name, since there will be faster calling conventions tomorow, but the ship has sailed. Cython uses METH_FASTCALL. It's too late to change it.

When I added METH_FASTCALL, I tried to keep it secret, but Cython started to use the private C API, and here we are :-)

**Comment by encukou** [comment]
I wouldn't worry that we can't come up with faster-sounding names. Like VictorCall, but that's already taken :)

**Comment by vstinner** [comment]
METH_V**I**CTORCALL sounds like a good name :-D

**Comment by encukou** [comment]
The `Timeout` in `test_peg_generator` doesn't look related. Will check the next build, https://buildbot.python.org/all/#/builders/58/builds/341

**Pull request merged**